src/main/java/org/gridsuite/study/server/service/NetworkModificationTreeService.java

```diff
@@ -748,20 +747,26 @@ public UUID getParentNode(UUID nodeUuid, NodeType nodeType) {
         return parentNodeUuidOpt.get();
     }
 
+    @Transactional(readOnly = true)
     public Optional<UUID> getParentNodeUuid(UUID nodeUuid) {
-        NodeEntity nodeEntity = nodesRepository.findById(nodeUuid).orElseThrow(() -> new StudyException(ELEMENT_NOT_FOUND));
+        NodeEntity nodeEntity = getNodeEntity(nodeUuid);
         return (nodeEntity.getType() == NodeType.ROOT) ? Optional.empty() : Optional.of(nodeEntity.getParentNode().getIdNode());
     }
 
     @Transactional(readOnly = true)
-    public UUID doGetLastParentNodeBuilt(UUID nodeUuid) {
-        NodeEntity nodeEntity = nodesRepository.findById(nodeUuid).orElseThrow(() -> new StudyException(ELEMENT_NOT_FOUND));
+    public UUID doGetLastParentNodeBuiltUuid(UUID nodeUuid) {
+        NodeEntity nodeEntity = getNodeEntity(nodeUuid);
+        return doGetLastParentNodeBuilt(nodeEntity).getIdNode();
+    }
+
+    @Transactional(readOnly = true)
```

Comment:
is this annotation necessary here, since this method is only called by doGetLastParentNodeBuiltUuid, already annotated with Transactioinal ...

Reply:
I think it's better to let it in case someone call it from outside. Both methods are usable.